flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/data/util/DataFormatConverters.java

```diff
@@ -1138,7 +1138,7 @@ private ArrayData toBinaryArray(T[] value) {
 				}
 			}
 			reuseWriter.complete();
-			return reuseArray;
+			return reuseArray.copy();
```

Comment:
We need to perform a similar change to the `DataStructureConverter`. `DataFormatConverters` are deprecated soon.

Reply:
thanks for reminding

Reply:
Also fixed in ArrayObjectArrayConverter